openapi/social/social.yaml

```diff
@@ -275,21 +361,57 @@ components:
             messageText:
               type: string
               description: 'The text of the message you’re making. This text is subject to limitations depending on what service you are posting to, such as Tweet length for Twitter. '
-            mediaUrls:
-              description: 'Links to pieces of media you want on the message you’re sending. These can be images, videos, and gifs. Media are subject to limitations depending on the social network you are posting to.'
+            status:
+              type: string
+              enum:
+                - scheduled
+                - posted
+              description: 'A string representing the tag assigned to a post.'
+              example: 'SCHEDULED'
+            media:
               type: array
+              description: 'An array containing the media attached to the post.'
               items:
-                type: string
-            sendAt:
+                type: object
+                properties:
+                  type:
+                    type: string
+                    enum:
+                      - image
+                      - video
+                      - gif
+                    example: 'image'
+                  url:
+                    type: string
+                    example: 'https://www.example.com/image.jpg'
+            postedAt:
               type: string
               format: date-time
               description: |-
-                The time you wish your message to be sent out.
-
-                Defaults to 5 minutes after the Message is made, and must be at least 5 minutes after the Message is created.
-
+                A string representing the timestamp at which the post was posted.
+                
+                Null in case of posts that have not been posted but just scheduled yet.
+                
                 Times used must be in ISO 8601 format, UTC time.
               example: '2022-02-04T16:52:34.000Z'
+            scheduledAt:
+              type: string
+              format: date-time
+              description: |-
+                A string representing the timestamp at which the post is scheduled.
+                
+                Null in case of posts was posted directly without scheduling.
```

Comment:
```suggestion
                Null if the post was not scheduled to be posted.
```

Reply:
Is there a created field on the database table that we can use as a fallback when populating this field?

Reply:
Yes, for each post, we track created and scheduled and postedDate field. In case its an immediate post, the scheduled and postedDate is same

Reply:
Would it make sense to take out the comment about null and add some code when building the returned values?

```golang
resp.ScheduledAt = db.ScheduledAt
if resp.ScheduledAt == "" {
  resp.ScheduledAt = db.PostedDate
}
```